src/ts/components/touchcontroloverlay.ts

```diff
@@ -0,0 +1,175 @@
+
+  private touchControlEvents = {
+    onSingleClick: new EventDispatcher<TouchControlOverlay, NoArgs>(),
+    onDoubleClick: new EventDispatcher<TouchControlOverlay, NoArgs>(),
+    onSeekBackward: new EventDispatcher<TouchControlOverlay, NoArgs>(),
+    onSeekForward: new EventDispatcher<TouchControlOverlay, NoArgs>(),
+  };
+
+  private playbackToggleButton: SmallCenteredPlaybackToggleButton;
+
+  // true if the last tap on the overlay was less than 500msec ago
+  private couldBeDoubleTapping: Boolean;
+  private doubleTapTimeout: Timeout;
+
+  private latestTapPosition: ClickPosition = {x: -1, y: -1};
+
+  constructor(config: TouchControlOverlayConfig = {}) {
+    super(config);
+
+    this.playbackToggleButton = new SmallCenteredPlaybackToggleButton({
+      enterFullscreenOnInitialPlayback: Boolean(config.enterFullscreenOnInitialPlayback),
+    });
+
+    this.config = this.mergeConfig(config, {
+      cssClass: 'ui-touchcontrol-overlay',
+      acceptsTouchWithUiHidden: true,
+      seekTime: 10,
+      seekDoubleTapMargin: 15,
+      components: [this.playbackToggleButton],
+    }, this.config);
+  }
+
+  configure(player: PlayerAPI, uimanager: UIInstanceManager): void {
+    super.configure(player, uimanager);
+
+    this.doubleTapTimeout = new Timeout(500, () => {
+      this.couldBeDoubleTapping = false;
+    });
+
+    uimanager.onControlsHide.subscribe(() => {
+      this.playbackToggleButton.hide();
+    });
+
+    uimanager.onControlsShow.subscribe(() => {
+      this.playbackToggleButton.show();
+    });
+
+    let playerSeekTime = 0;
+
+    this.touchControlEvents.onSeekBackward.subscribe(() => {
+      playerSeekTime -= this.config.seekTime;
+      player.seek(playerSeekTime);
+    });
+
+    this.touchControlEvents.onSeekForward.subscribe(() => {
+      playerSeekTime += this.config.seekTime;
+      player.seek(playerSeekTime);
+    });
+
+    this.touchControlEvents.onSingleClick.subscribe((_, e) => {
+      uimanager.getUI().toggleUiShown();
+      playerSeekTime = player.getCurrentTime();
+
+      const eventTarget = (e as Event).target as HTMLElementWithComponent;
+      const rect = eventTarget.getBoundingClientRect();
+      const eventTapX = ((<any>e).clientX) - rect.left;
+      const eventTapY = ((<any>e).clientY) - rect.top;
+      this.latestTapPosition = {x: eventTapX, y: eventTapY};
+    });
+
+    this.touchControlEvents.onDoubleClick.subscribe((_, e) => {
+      uimanager.getUI().hideUi();
+      const event = e as Event;
+      const eventTarget = event.target as HTMLElementWithComponent;
+      if (!eventTarget || !((eventTarget).component instanceof TouchControlOverlay)) {
+        return;
+      }
+
+      const width = eventTarget.clientWidth;
+      const tapMargin = width * 0.4;
+      const rect = eventTarget.getBoundingClientRect();
+      const eventTapX = ((<any>e).clientX) - rect.left;
+      const eventTapY = ((<any>e).clientY) - rect.top;
```

Comment:
Same here, there should be something better than `any`.
```suggestion
      const eventTapX = ((e as MouseEvent).clientX) - rect.left;
      const eventTapY = ((e as MouseEvent).clientY) - rect.top;
```